pos_sale_order/models/pos_session.py

```diff
@@ -83,29 +92,43 @@ def _create_bank_statement_line(self, statement, payments):
                 _("Following sales are not invoiced, please invoice it: \n- %s")
                 % ("\n- ".join(sale_missing_invoice.mapped("name")))
             )
-        move_lines = (
-            bk_line.move_id.line_ids
-            + payments.pos_sale_order_id.invoice_ids.filtered(
-                lambda s: s.state == "posted"
-            ).line_ids
+        return payments.pos_sale_order_id.invoice_ids.filtered(
+            # skip canceled invoice
+            lambda s: s.state
+            == "posted"
+        ).line_ids.filtered(
+            # get receivable line
+            lambda s: s.account_id
+            == payments.payment_method_id.receivable_account_id
         )
-        return move_lines.filtered(
+
+    def _create_bank_statement_line(self, statement, payments, record_ref):
+        vals = self._prepare_sale_statement(payments, record_ref)
```

Comment:
Only one argument here
```suggestion
        vals = self._prepare_sale_statement(payments)
```